migrate/v0_15/migrate.go

```diff
@@ -141,6 +148,96 @@ func MigrateAccounts(genesisState auth.GenesisState, genesisTime time.Time) auth
+			err := acc.SetCoins(acc.GetCoins().Add(sdk.NewCoin("swp", sdk.NewInt(137500000e6))))
+			if err != nil {
+				panic(err)
+			}
+			accounts[i] = authexported.GenesisAccount(acc)
+		}
+
+	}
+
+	// Add ecosystem account
+	swpEcosystemAddr := mustAccAddressFromBech32("kava174crgzk006lj39s258gjxkh0p9tmmq8ugdcg0n")
+	swpEcosystemBacc := auth.NewBaseAccountWithAddress(swpEcosystemAddr)
+	// 11.5M because 1M has already been airdropped
+	ecoSystemCoins := sdk.NewCoin("swp", sdk.NewInt(11500000000000))
+	err := swpEcosystemBacc.SetCoins(sdk.NewCoins(ecoSystemCoins))
+	if err != nil {
+		panic(err)
+	}
+
+	// Add team vesting account
+	swpTeamAddr := mustAccAddressFromBech32("kava129eqnykzkc5ceyq9sv7ltxev22y8qwm94kr0ew")
+	swpTeamBacc := auth.NewBaseAccountWithAddress(swpTeamAddr)
+	teamCoins := sdk.NewCoin("swp", sdk.NewInt(37500000000000))
+	err = swpTeamBacc.SetCoins(sdk.NewCoins(teamCoins))
+	if err != nil {
+		panic(err)
+	}
+	swpTeamBva, err := vesting.NewBaseVestingAccount(&swpTeamBacc, sdk.NewCoins(teamCoins), 1693407600)
+	if err != nil {
+		panic(err)
+	}
+	swpTeamTokenSchedule := []sdk.Coin{
+		sdk.NewCoin("swp", sdk.NewInt(18750000000000)),
+		sdk.NewCoin("swp", sdk.NewInt(4687500000000)),
+		sdk.NewCoin("swp", sdk.NewInt(4687500000000)),
+		sdk.NewCoin("swp", sdk.NewInt(4687500000000)),
+		sdk.NewCoin("swp", sdk.NewInt(4687500000000)),
+	}
+
+	swpTeamVestingLengths := []int64{31536000, 7948800, 7776000, 7948800, 7862400}
+	swpTeamVestingPeriods := vesting.Periods{}
+	for i, vestingCoin := range swpTeamTokenSchedule {
+		period := vesting.Period{Length: swpTeamVestingLengths[i], Amount: sdk.NewCoins(vestingCoin)}
+		swpTeamVestingPeriods = append(swpTeamVestingPeriods, period)
+	}
+	swpTeamVestingAccount := vesting.NewPeriodicVestingAccountRaw(swpTeamBva, GenesisTime.Unix(), swpTeamVestingPeriods)
+
+	// Add treasury vesting account
+	swpTreasuryAddr := mustAccAddressFromBech32("kava1w56wrusdnrv4tvn86eyam65wwhqatmsqg9fxjm")
+	swpTreasuryBacc := auth.NewBaseAccountWithAddress(swpTreasuryAddr)
+	swpTreasuryCoins := sdk.NewCoin("swp", sdk.NewInt(62500000000000))
+	err = swpTreasuryBacc.SetCoins(sdk.NewCoins(swpTreasuryCoins))
+	if err != nil {
+		panic(err)
+	}
+	swpTreasuryOriginalVesting := sdk.NewCoin("swp", sdk.NewInt(46875000000000))
+	swpTreasuryBva, err := vesting.NewBaseVestingAccount(&swpTreasuryBacc, sdk.NewCoins(swpTreasuryOriginalVesting), 1693407600)
+	if err != nil {
+		panic(err)
+	}
+	swpTreasuryTokenSchedule := []sdk.Coin{
+		sdk.NewCoin("swp", sdk.NewInt(5859375e6)),
+		sdk.NewCoin("swp", sdk.NewInt(5859375e6)),
+		sdk.NewCoin("swp", sdk.NewInt(5859375e6)),
+		sdk.NewCoin("swp", sdk.NewInt(5859375e6)),
+		sdk.NewCoin("swp", sdk.NewInt(5859375e6)),
+		sdk.NewCoin("swp", sdk.NewInt(5859375e6)),
+		sdk.NewCoin("swp", sdk.NewInt(5859375e6)),
+		sdk.NewCoin("swp", sdk.NewInt(5859375e6)),
+	}
+	swpTreasuryVestingLengths := []int64{7948800, 7776000, 7948800, 7862400, 7948800, 7776000, 7948800, 7862400}
```

Comment:
Why is there variation in treasury vesting lengths when the token amount for each is the same?

Reply:
Vesting dates fall at the end of the month. So for months like Feb, vesting lengths are a bit shorter, etc